dbfit-java/core/src/test/java/dbfit/util/DbParameterAccessorTest.java

```diff
@@ -0,0 +1,53 @@
+package dbfit.util;
+
+import dbfit.fixture.StatementExecution;
+import org.junit.Test;
+import static org.mockito.Mockito.*;
+import java.sql.Types;
+
+public class DbParameterAccessorTest {
+
+    @Test
+    public void inputObjectIsTransformedBeforeJdbcBindingWhenTransformerSpecifierTest() throws Exception {
+        int sqlType = Types.VARCHAR;
+        Class<?> javaType = String.class;
+        DbParameterAccessor dbpa;
+        String inputValue = "The input value";
+        int position = 1;
+        Integer outputValue = 5;
+        String userDefinedTypeName = "whatever";
+
+        TypeTransformer tt = mock(TypeTransformer.class);
+        when(tt.transform(inputValue)).thenReturn(outputValue);
+
+        TypeTransformerFactory ttf = mock(TypeTransformerFactory.class);
+        when(ttf.getTransformer(inputValue.getClass())).thenReturn(tt);
+
+        StatementExecution se = mock(StatementExecution.class);
+
+        dbpa = new DbParameterAccessor("dummy", Direction.INPUT_OUTPUT, sqlType, userDefinedTypeName, javaType, position, ttf);
+        dbpa.bindTo(se, position);
+        dbpa.set(inputValue);
+        verify(se).setObject(position, outputValue, sqlType, userDefinedTypeName);
+    }
+
+    @Test
+    public void inputObjectIsNotTransformedBeforeJdbcBindingWhenNoTransformerSpecifierTest() throws Exception {
+        int sqlType = Types.VARCHAR;
+        Class<?> javaType = String.class;
+        DbParameterAccessor dbpa;
+        String inputValue = "The input value";
+        int position = 1;
+        String userDefinedTypeName = "whatever";
+
+        TypeTransformerFactory ttf = mock(TypeTransformerFactory.class);
+        when(ttf.getTransformer(inputValue.getClass())).thenReturn(null);
+
+        StatementExecution se = mock(StatementExecution.class);
+
+        dbpa = new DbParameterAccessor("dummy", Direction.INPUT_OUTPUT, sqlType, userDefinedTypeName, javaType, position, ttf);
+        dbpa.bindTo(se, position);
+        dbpa.set(inputValue);
+        verify(se).setObject(position, inputValue, sqlType, userDefinedTypeName);
+    }
+}
```

Comment:
I'm not sure how else I can check to see what the `set()` call has actually done.

Where else can I intercept things using the Mockito bits?